Core/src/Propagator/detail/CovarianceEngine.cpp

```diff
@@ -157,13 +157,16 @@ void reinitializeJacobians(
   jacobianLocalToGlobal = BoundToFreeMatrix::Zero();
 
   // Reset the jacobian from local to global
-  Vector2D loc{0., 0.};
   const Vector3D position = parameters.segment<3>(eFreePos0);
   const Vector3D direction = parameters.segment<3>(eFreeDir0);
-  surface.globalToLocal(geoContext, position, direction, loc);
+  auto loc = surface.globalToLocal(geoContext, position, direction);
+  if (not loc.ok()) {
+    std::cout << " GLOBAL TO LOCAL NOT OK! " << std::endl;
```

Comment:
That was actually an accident, should. not have been committed like this, but I guess we should use a local logger for this.

Reply:
In the propagation you should have a logger instance available from the prop options.

Reply:
Oh I see, at least I can shout where the inconsistency appears, which is good for debugging. I will have a look.

Reply:
Actually the `CovarianceEngine` didn't have access to this, but I created a local logger only in case it fails to tell you where.

Reply:
I was going to comment that this could live in `CovarianceEngine` as well, but that is not even a type, or even it's own namespace. Bit weird. Resolving this.